cmd/serve_test.go

```diff
@@ -12,9 +13,17 @@ import (
 	. "github.com/onsi/gomega"
 )
 
+const (
+	basePort = 5000
```

Comment:
Not sure if it's happening in practice, but using a predefined port could be a cause of flakiness if something else already uses it, or even a previous blocky run made it unavailable for a bit.  
From a quick look it seems possible to use port 0 and then use a listener which can give you the actual port via `.Addr()`. `http.ListenAndServe` would need to be replaced with creating a server and passing the listener to `.Serve`, like we do in `server.go`.  
I guess making a custom `listenAndServe` func to wrap that would make it easier.

Reply:
That's why I created a helper function which adds the test process number to the base port.
This way the parallel running tests don't block each others port.

Reply:
Ah my bad I missed that!

Reply:
Turns out I didn't review the full changes, just one push I think!
Too late I guess :stuck_out_tongue: